### ISSUE TITLE
RGBW colors

### ISSUE THREAD

**Issue by rbhun** [bokub/rgb-light-card#84]
Hi

just wanted to add, RGBW leds are working too, like this:
colors:
  - rgbw_color:
      - 200
      - 0
      - 0
      - 100
    brightness: 155
however, the icon color does not take into account the white channel, so a 0,0,0,100 would be black instead of white. I fixed that with adding icon_color: white, but it would be better representation of colours if the W channel is added.

**Comment by bokub** [comment]
That's right. Do you have any algorithm or code to suggest, that could calculate the icon color based on a RGBW or RGBWW array ?

Thanks

Also, duplicate of #79

**Comment by j9brown** [comment]
For RGB/W and RGB/WW/CW a simple expedient would be to choose an RGB to represent warm white and another one for cold white.  Scale the appropriate WW and CW colors by the brightness of their color channels and add them to the RGB color of the remaining channels.  If the resulting RGB color has components greater than 100% then divide all components by the largest component to normalize it.

For more general color temperature to RGB, check out this algorithm (which I have used before in another project): https://tannerhelland.com/2012/09/18/convert-temperature-rgb-algorithm-code.html

You can also use the same algorithm to choose the representative WW and CW colors given an assumed (or provided) color temperature for each channel.

**Comment by j9brown** [comment]
Here is the exact thing you need, already in the Home Assistant front end: https://github.com/home-assistant/frontend/blob/0d97afb3f2c6a04f64443c49b350567d1ac971f4/src/common/color/convert-light-color.ts#L67
